### DISCUSSION TITLE
Difficulty in deploying react app on GitHubPages

### DISCUSSION THREAD

**Discussion by Karan-Ghatt** [community/community#23770] [Other Feature Feedback, Questions, & Ideas]
<p>Repo: <a href="https://github.com/Karan-Ghatt/new-app" rel="nofollow noopener">https://github.com/Karan-Ghatt/new-app</a><br>
Website: <a href="https://karan-ghatt.github.io/new-app/" rel="nofollow noopener">https://karan-ghatt.github.io/new-app/</a></p>
<p>I’m having some difficulty in deploying a simple react app.<br>
I went through the steps of creating my repo, going though the whole npm run build and npm deploy, I made sure to change the relevant aspects of the package.json file.</p>
<p>So I manage to get the page hosted through GitHubPages, but my JS. doesn’t seem to run! There’s meant to be a whole visual element on the page with buttons, but alas, they are no where to be found.</p>
<p>I’ve check the .chunks.js in static/.js files referenced in the index.html file, and it seems to have my code for the JS elements, but it again, it dose not appear on the page.</p>
<p>It’s like 2am and I could really do with some help please!!</p>
<p>Thank you!</p>

**Comment by darrowby385** [comment, marked as answer]
<p>Hi, <a class="mention" href="/karan-ghatt">@Karan-Ghatt</a>!</p>
<p>It looks like the trouble here might be due to some broken URLs in your HTML. For example, this line of your <code>index.html</code> file includes the value <code>href="/Karan-Ghatt/new-app/manifest.json"</code>:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/Karan-Ghatt/new-app/blob/293841b92b6bb0c1ffd5a2aaf918a82e6728d504/index.html#L11" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/Karan-Ghatt/new-app/blob/293841b92b6bb0c1ffd5a2aaf918a82e6728d504/index.html#L11" target="_blank" rel="nofollow noopener">Karan-Ghatt/new-app/blob/293841b92b6bb0c1ffd5a2aaf918a82e6728d504/index.html#L11</a></h4>
<pre class="onebox"><code class="lang-html"><ol class="start lines" start="1" style="counter-reset: li-counter 0 ;">
<li>&lt;!doctype html&gt;</li>
<li>&lt;html lang="en"&gt;</li>
<li>
</li>
<li>&lt;head&gt;</li>
<li>    &lt;meta charset="utf-8" /&gt;</li>
<li>    &lt;link rel="icon" href="logo.png" /&gt;</li>
<li>    &lt;meta name="viewport" content="width=device-width,initial-scale=1" /&gt;</li>
<li>    &lt;meta name="theme-color" content="#000000" /&gt;</li>
<li>    &lt;meta name="description" content="Web site created using create-react-app" /&gt;</li>
<li>    &lt;link rel="apple-touch-icon" href="logo.png" /&gt;</li>
<li class="selected">    &lt;link rel="manifest" href="/Karan-Ghatt/new-app/manifest.json" /&gt;</li>
<li>    &lt;title&gt;Visual Sorting&lt;/title&gt;</li>
<li>    &lt;link href="/Karan-Ghatt/new-app/static/css/main.f2084247.chunk.css" rel="stylesheet"&gt;</li>
<li>&lt;/head&gt;</li>
<li>
</li>
<li>&lt;body&gt;</li>
<li>    &lt;style&gt;</li>
<li>        ul {</li>
<li>            list-style-type: none;</li>
<li>            margin: 0;</li>
<li>            padding: 0;</li>
</ol></code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Your site is published at <code>https://karan-ghatt.github.io/new-app/</code>, not <code>https://karan-ghatt.github.io/Karan-Ghatt/new-app/</code>, as the above URL implies. Correcting these broken URLs, likely through a single configuration value in your React project, should resolve this trouble.</p>
<p>As a side note, I uncovered this trouble by looking for 404s in the Chrome developer console, while viewing your published site. If you’re not familiar with the Chrome developer console, I’d highly recommend checking it out, as it’s a great tool for debugging exactly this sort of trouble:</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/19504141/181079271-38c1e97c-6f78-4e86-b124-4933ce7c02c1.png" class="site-icon" width="32" height="32">
      <a href="https://developers.google.com/web/tools/chrome-devtools/console/" target="_blank" rel="nofollow noopener">Google Developers</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/388;"><img src="https://user-images.githubusercontent.com/19504141/181079272-aee3babe-2d59-4ac2-a4f4-ed4f294bb033.png" class="thumbnail" width="690" height="388" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/037e38456d9b51512ec50d968a36277eb89f28f2_2_10x10.png"></div>

<h3><a href="https://developers.google.com/web/tools/chrome-devtools/console/" target="_blank" rel="nofollow noopener">Console Overview  |  Tools for Web Developers  |  Google...</a></h3>

<p>The main uses of the Chrome DevTools Console are logging messages and running JavaScript.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I hope that helps!</p>
<p>Friday</p>

**Comment by Karan-Ghatt** [comment]
<p>Thank you for the insight!!<br>
I’m still a little new to react, I’m not too sure what you mean by configuration value?</p>
<p>Could you help me whats wrong with the URLs above, what is it that makes them broken?</p>
<p>I apprentice this may be pretty simple stuff, so thanks again for your time!</p>
